client/state/data-layer/wpcom/read/teams/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import { READER_TEAMS_REQUEST, READER_TEAMS_RECEIVE } from 'state/action-types';
+import wpcom from 'lib/wp';
+
+export function handleTeamsRequest( store, action, next ) {
+	wpcom.req.get( '/read/teams', { apiVersion: '1.2' } )
+		.then(
+			payload => {
```

Comment:
My thoughts:
1. naming it `payload` is fine
2. there should be api response validation.  something like:

```jsx
function validateApi( payload ) {
   // make sure payload has the correct shape
}
```